Chap_Terms.tex

```diff
@@ -4,39 +4,30 @@
+The following terms and terminology is used throughout this document:
 \begin{itemize}
-\item \declareterm{session}\emph{session} refers to an allocated set of resources assigned to a particular user by the system \ac{WLM}. Historically, \ac{HPC} sessions have consisted of a static allocation of resources - i.e., a block of resources are assigned to a user in response to a specific request and managed as a unified collection. However, this is changing in response to the growing use of dynamic programming models that require on-the-fly allocation and release of system resources. Accordingly, the term \emph{session} in this document refers to the current block of assigned resources and is a potentially dynamic entity.
-\item \declareterm{slot}\declareterm{slots}\emph{slot} refers to an allocated entry for a process. \acp{WLM} frequently allocate entire nodes to a \emph{session}, but can also be configured to define the maximum number of processes that can simultaneously be executed on each node. This often corresponds to the number of hardware \acp{PU} (typically cores, but can also be defined as hardware threads) on the node. However, the correlation between hardware \acp{PU} and slot allocations strictly depends upon system configuration.
-\item \declareterm{job}\emph{job} refers to a set of one or more \emph{applications} executed as a single invocation by the user within a session. For example, ``\textit{mpiexec -n 1 app1 : -n 2 app2}'' is considered a single \ac{MPMD} job containing two applications.
+\item \declareterm{session}\emph{session} refers to an allocated set of resources assigned to a particular user by the system \ac{WLM}. Historically, \ac{HPC} \emph{sessions} have consisted of a static allocation of resources - i.e., a block of resources are assigned to a user in response to a specific request and managed as a unified collection. However, this is changing in response to the growing use of dynamic programming models that require on-the-fly allocation and release of system resources. Accordingly, the term \emph{session} in this document refers to the current block of assigned resources and is a potentially dynamic entity.
+\item \declareterm{slot}\declareterm{slots}\emph{slot} refers to an allocated set of resources assigned to a process. \acp{WLM} frequently allocate entire nodes to a \emph{session}, but can also be configured to define the maximum number of processes that can simultaneously be executed on each node. This often corresponds to the number of hardware \acp{PU} (typically cores, but can also be defined as hardware threads) on the node. However, the correlation between hardware \acp{PU} and slot allocations strictly depends upon system configuration.
+\item \declareterm{application}\emph{application} refers to a single executable (binary, script, etc.) member of a \emph{job}. Applications consist of one or more \emph{processes}, either operating independently or cooperatively at any given time during their execution.
```

Comment:
The statements "applications refer to binaries" and "application consit of ... processes" doesn't make sense to me: the first is static, the second dynamic. Does this refer to the actual code or the execution of such code?

Reply:
It's the grouping of processes in the MPMD job. The binary name could be the distinguishing factor. I see your point about mixing binary and process here - it could be clearer.

Reply:
Binary name is yet a different term that has not been introduced, yet, and could lead to further complications? Also, some binaries may be created on the fly or modified on the fly - what does this mean for the term application. Two parts may run independently using the MPI Sessions model in their subsets (i.e., multiple applications) and later during the execution decide to create a super session (i.e., turning into one application). You could have a tool on a separate binary name, but linked against the same MPI library, that is integrated into one application, although it does something definitely different? What about if dynamic libraries change (they could even have different names, worst case)? I still think the term application is ill-defined, as this can mean anything from a user's point of view.